Utils/hdinsight-node-common/src/com/microsoft/azure/hdinsight/common/task/YarnHistoryTask.java

```diff
@@ -40,29 +44,32 @@
 
     protected final IClusterDetail clusterDetail;
     protected final String path;
-    private final CredentialsProvider credentialsProvider =  new BasicCredentialsProvider();
-
+    private static final CredentialsProvider CREDENTIALS_PROVIDER =  new BasicCredentialsProvider();
```

Comment:
Instance would be better here, it would help the testing in future. And the static limits that all yarn history task instance would share the same basic type credential provider.